packages/trader/src/Modules/Trading/Components/Form/TradeParams/amount-mobile.tsx

```diff
@@ -17,28 +35,31 @@ const Basis = observer(
         selected_basis,
         setSelectedAmount,
         setAmountError,
-    }) => {
+    }: TBasis) => {
         const { ui, client } = useStore();
         const { addToast, vanilla_trade_type } = ui;
         const { currency } = client;
         const {
             onChangeMultiple,
-            trade_amount,
-            trade_basis,
-            trade_duration_unit,
-            trade_duration,
+            amount: trade_amount,
+            basis: trade_basis,
+            duration_unit: trade_duration_unit,
+            duration: trade_duration,
             contract_type,
             stake_boundary,
         } = useTraderStore();
+
         const user_currency_decimal_places = getDecimalPlaces(currency);
-        const onNumberChange = num => {
+        const onNumberChange = (num: number | string) => {
             setSelectedAmount(basis, num);
             validateAmount(num);
         };
-        const formatAmount = value =>
-            !isNaN(value) && value !== '' ? Number(value).toFixed(user_currency_decimal_places) : value;
-        const setBasisAndAmount = amount => {
-            const on_change_obj = {};
+        const formatAmount = (value: number | string): number => {
+            const numericValue = typeof value === 'string' ? Number(value) : value;
+            return !isNaN(numericValue) ? parseFloat(numericValue.toFixed(user_currency_decimal_places)) : numericValue;
+        };
```

Comment:
```suggestion
        const formatAmount = (value: number | string) =>
            !isNaN(+value) && value !== '' ? Number(value).toFixed(user_currency_decimal_places) : value;
```
@henry-deriv please also update `format` callback return type in packages/components/src/components/numpad/numpad.tsx at line 26 from number to number | string and also, `format` prop should be non-required (?): 
```
format?: (v: string) => number | string;
```

Reply:
yes already done